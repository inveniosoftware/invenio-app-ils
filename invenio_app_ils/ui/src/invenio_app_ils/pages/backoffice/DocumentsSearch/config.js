import { invenioConfig } from '../../../common/config';
import _capitalize from 'lodash/capitalize';

const resultsPerPageValues = [
  {
    text: '10',
    value: 10,
  },
  {
    text: '20',
    value: 20,
  },
  {
    text: '50',
    value: 50,
  },
];

const config = invenioConfig.documents.search;
const sortByValues = config.sortBy.values.map(sortField => {
  return { text: sortField.title, value: sortField.field };
});
const sortByValueOnEmptyQuery = config.sortBy.onEmptyQuery;
const sortOrderValues = config.sortOrder.map(sortField => {
  return { text: _capitalize(sortField), value: sortField };
});

const aggsMappings = {
  document_types: 'Document types',
  keywords: 'Keywords',
  languages: 'Languages',
  items_available_for_loan: 'Available items',
};
const aggs = config.aggs.map(agg => {
  return { title: aggsMappings[agg], field: agg };
});

export default {
  RESULTS_PER_PAGE: resultsPerPageValues,
  SORT_BY: sortByValues,
  SORT_BY_ON_EMPTY_QUERY: sortByValueOnEmptyQuery,
  SORT_ORDER: sortOrderValues,
  AGGREGATIONS: aggs,
};
