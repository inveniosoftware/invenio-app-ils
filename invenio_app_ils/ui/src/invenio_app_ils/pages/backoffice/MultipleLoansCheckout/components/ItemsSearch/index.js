import { compose } from 'redux';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import ItemsSearchComponent from './ItemsSearch';
import { clearResults, fetchItems, updateQueryString } from './state/actions';
import { addItemToBasket } from '../ItemsBasket/state/actions';

const mapDispatchToProps = dispatch => ({
  fetchItems: barcode => dispatch(fetchItems(barcode)),
  updateQueryString: qs => dispatch(updateQueryString(qs)),
  addItemToBasket: item => dispatch(addItemToBasket(item)),
  clearResults: () => dispatch(clearResults()),
});

const mapStateToProps = state => ({
  isLoading: state.selectedUser.isLoading,
  items: state.itemsSearchInput.data,
  hasError: state.itemsSearchInput.hasError,
  queryString: state.itemsSearchInput.multiCheckoutQueryString,
});

export const ItemsSearch = compose(
  withRouter,
  connect(
    mapStateToProps,
    mapDispatchToProps
  )
)(ItemsSearchComponent);
