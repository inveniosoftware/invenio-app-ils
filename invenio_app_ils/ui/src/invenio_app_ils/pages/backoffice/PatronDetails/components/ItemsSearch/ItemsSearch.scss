.patron-items {
  .results-list {
    margin: 0 !important;
  }

  .sub.header {
    line-height: 0em;
    padding: 0 0 0.5em 0;
    margin: 0;
    color: rgba(0, 0, 0, 0.6);
  }

  .search-bar {
    padding: 1em 0;
  }
}
