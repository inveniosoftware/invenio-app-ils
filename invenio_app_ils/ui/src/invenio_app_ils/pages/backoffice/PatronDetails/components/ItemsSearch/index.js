import { compose } from 'redux';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import ItemsSearchComponent from './ItemsSearch';
import {
  clearResults,
  fetchItems,
  fetchUpdatedCurrentLoans,
  updateQueryString,
} from './state/actions';
import { checkoutItem } from '../ItemsCheckout/state/actions';

const mapDispatchToProps = dispatch => ({
  fetchItems: barcode => dispatch(fetchItems(barcode)),
  updateQueryString: qs => dispatch(updateQueryString(qs)),
  clearResults: () => dispatch(clearResults()),
  checkoutItem: (item, patronPid, shouldForceCheckout) =>
    dispatch(checkoutItem(item, patronPid, shouldForceCheckout)),
  fetchUpdatedCurrentLoans: patronPid =>
    dispatch(fetchUpdatedCurrentLoans(patronPid)),
});

const mapStateToProps = state => ({
  isLoading: state.patronDetails.isLoading,
  items: state.itemsSearchInput.data,
  hasError: state.itemsSearchInput.hasError,
  queryString: state.itemsSearchInput.itemCheckoutQueryString,
});

export const ItemsSearch = compose(
  withRouter,
  connect(
    mapStateToProps,
    mapDispatchToProps
  )
)(ItemsSearchComponent);
