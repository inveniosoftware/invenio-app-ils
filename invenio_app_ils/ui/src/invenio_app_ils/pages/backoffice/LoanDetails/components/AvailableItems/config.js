import { invenioConfig } from '../../../common/config';

export default invenioConfig.items.available;
