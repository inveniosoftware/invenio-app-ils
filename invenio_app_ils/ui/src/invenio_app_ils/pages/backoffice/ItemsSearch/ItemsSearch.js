import React, { Component } from 'react';
import { Grid, Item, Segment, Icon, Header } from 'semantic-ui-react';
import {
  ReactSearchKit,
  SearchBar,
  ResultsList,
  ResultsLoader,
  EmptyResults,
  Error,
  Pagination,
  Count,
  Aggregator,
  InvenioSearchApi,
} from 'react-searchkit';
import { getSearchConfig } from '../../../common/config';
import {
  Error as IlsError,
  SearchBar as ItemsSearchBar,
  ResultsSort,
} from '../../../common/components';
import { item as itemApi } from '../../../common/api';
import { ExportReactSearchKitResults, ItemListEntry } from '../components';
import { ClearButton, NewButton } from '../components/buttons';
import { BackOfficeRoutes } from '../../../routes/urls';

export class ItemsSearch extends Component {
  searchApi = new InvenioSearchApi({
    url: itemApi.searchBaseURL,
    withCredentials: true,
  });
  searchConfig = getSearchConfig('items');

  renderSearchBar = (_, queryString, onInputChange, executeSearch) => {
    return (
      <ItemsSearchBar
        currentQueryString={queryString}
        onInputChange={onInputChange}
        executeSearch={executeSearch}
        placeholder={'Search for items'}
      />
    );
  };

  renderEmptyResults = (queryString, resetQuery) => {
    return (
      <Segment placeholder textAlign="center">
        <Header icon>
          <Icon name="search" />
          No items found!
        </Header>
        <div>Current search "{queryString}"</div>
        <Segment.Inline>
          <ClearButton
            clickHandler={() => {
              resetQuery();
            }}
          />
          <NewButton text={'New item'} to={BackOfficeRoutes.itemCreate} />
        </Segment.Inline>
      </Segment>
    );
  };

  renderError = error => {
    return <IlsError error={error} />;
  };

  renderCount = totalResults => {
    return <div>{totalResults} results</div>;
  };

  renderAggregations = () => {
    const components = this.searchConfig.AGGREGATIONS.map(agg => (
      <div key={agg.field}>
        <Aggregator title={agg.title} field={agg.field} />
      </div>
    ));
    return (
      <>
        <NewButton fluid text={'New item'} to={BackOfficeRoutes.itemCreate} />
        {components}
      </>
    );
  };

  renderHeader = () => {
    return (
      <Grid columns={2}>
        <Grid.Column>
          <Count renderElement={this.renderCount} />
        </Grid.Column>
        <Grid.Column textAlign="right">
          <ResultsSort searchConfig={this.searchConfig} />
        </Grid.Column>
      </Grid>
    );
  };

  renderFooter = () => {
    return (
      <Grid columns={3}>
        <Grid.Column width={5} />
        <Grid.Column width={6}>
          <Pagination />
        </Grid.Column>
        <Grid.Column width={5} textAlign="right">
          <ExportReactSearchKitResults exportBaseUrl={itemApi.searchBaseURL} />
        </Grid.Column>
      </Grid>
    );
  };

  renderItemList = results => {
    const entries = results.map(res => (
      <ItemListEntry item={res} key={res.id} />
    ));
    return <Item.Group>{entries}</Item.Group>;
  };

  render() {
    return (
      <ReactSearchKit searchApi={this.searchApi}>
        <Grid>
          <Grid.Row columns={1}>
            <Grid.Column>
              <SearchBar renderElement={this.renderSearchBar} />
            </Grid.Column>
          </Grid.Row>

          <Grid.Row columns={2}>
            <ResultsLoader>
              <Grid.Column width={3}>{this.renderAggregations()}</Grid.Column>
              <Grid.Column width={13}>
                <EmptyResults renderElement={this.renderEmptyResults} />
                <Error renderElement={this.renderError} />
                {this.renderHeader()}
                <ResultsList renderElement={this.renderItemList} />
                {this.renderFooter()}
              </Grid.Column>
            </ResultsLoader>
          </Grid.Row>
        </Grid>
      </ReactSearchKit>
    );
  }
}
