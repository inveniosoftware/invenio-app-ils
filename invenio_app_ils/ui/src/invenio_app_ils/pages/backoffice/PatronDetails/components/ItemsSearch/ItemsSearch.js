import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { generatePath } from 'react-router';
import {
  Container,
  Grid,
  Segment,
  Icon,
  Header,
  Button,
  Input,
} from 'semantic-ui-react';
import { BackOfficeURLS } from '../../../../../common/urls';
import { Error, Loader } from '../../../../../common/components';
import { ResultsList as ItemsResultsList } from './components';
import _isEmpty from 'lodash/isEmpty';
import './ItemsSearch.scss';

export default class ItemsSearch extends Component {
  constructor(props) {
    super(props);
    this.fetchItems = this.props.fetchItems;
    this.updateQueryString = this.props.updateQueryString;
    this.clearResults = this.props.clearResults;
    this.checkoutItem = this.props.checkoutItem;
    this.fetchUpdatedCurrentLoans = this.props.fetchUpdatedCurrentLoans;
    this.state = { prevSearchQuery: '' };
  }

  componentDidMount() {
    if (this.searchInput) {
      this.searchInput.focus();
    }
  }

  onInputChange = queryString => {
    this.updateQueryString(queryString);
  };

  executeSearch = queryString => {
    queryString = queryString || this.props.queryString;
    this.setState({ prevSearchQuery: queryString });

    return this.fetchItems(queryString);
  };

  _onPasteHandler = async event => {
    let queryString = event.clipboardData.getData('Text');

    if (queryString) {
      await this.executeSearch(queryString);

      const { hits } = this.props.items;
      const hasOneHit =
        !_isEmpty(hits) &&
        hits.length === 1 &&
        hits[0].metadata.status === 'CAN_CIRCULATE';
      if (hasOneHit) {
        await this.checkoutItem(hits[0], this.props.patron);
        this.clearResults();
        this.fetchUpdatedCurrentLoans(this.props.patron);
        this.setState({ prevSearchQuery: '' });
      }
    }
  };

  _onKeyPressHandler = e => {
    if (e.key === 'Enter' && this.props.queryString) {
      this.executeSearch();
    }
  };

  _renderSearchBar = () => {
    return (
      <Input
        action={{
          content: 'Search',
          onClick: () => {
            this.executeSearch();
          },
        }}
        fluid
        focus
        placeholder={'Search by barcode...'}
        onChange={(e, { value }) => {
          this.onInputChange(value);
        }}
        onPaste={e => {
          this._onPasteHandler(e);
        }}
        value={this.props.queryString}
        onKeyPress={event => {
          this._onKeyPressHandler(event);
        }}
        ref={input => {
          this.searchInput = input;
        }}
      />
    );
  };

  _renderResultsList = results => {
    return (
      <div className="results-list">
        {this._renderHeader(results.hits.length)}
        <ItemsResultsList
          patron={this.props.patron}
          clearResults={this.clearResults}
          results={results}
          checkoutItem={this.checkoutItem}
          fetchPatronCurrentLoans={this.fetchUpdatedCurrentLoans}
          viewDetailsClickHandler={itemPid => {
            const path = generatePath(BackOfficeURLS.itemDetails, {
              itemPid: itemPid,
            });
            this.props.history.push(path);
          }}
        />
      </div>
    );
  };

  _renderEmptyResults = (queryString, resetQuery) => {
    return (
      <Segment placeholder textAlign="center">
        <Header icon>
          <Icon name="search" />
          Found no items matching this barcode.
        </Header>
        <div className="empty-results-current">
          Current search phrase "{queryString}"
        </div>
        <Segment.Inline>
          <Button primary onClick={() => resetQuery('')}>
            Clear query
          </Button>
        </Segment.Inline>
      </Segment>
    );
  };

  _renderHeader = totalResults => {
    return <p>Found {totalResults} item(s).</p>;
  };

  render() {
    const { items, isLoading, error } = this.props;
    const { prevSearchQuery } = this.state;
    return (
      <Segment className={'patron-items'}>
        <Header as={'h3'}>Items</Header>
        <Header.Subheader>Search items by barcode.</Header.Subheader>
        <Container className={'search-bar'}>
          {this._renderSearchBar()}
        </Container>
        <Grid columns={1} stackable relaxed className="items-search-container">
          <Grid.Column width={16}>
            <Loader isLoading={isLoading}>
              <Error error={error}>
                {!_isEmpty(items.hits)
                  ? this._renderResultsList(items)
                  : this._renderEmptyResults(
                      prevSearchQuery,
                      this.updateQueryString
                    )}
              </Error>
            </Loader>
          </Grid.Column>
        </Grid>
      </Segment>
    );
  }
}

ItemsSearch.propTypes = {
  queryString: PropTypes.string,
  updateQueryString: PropTypes.func.isRequired,
  items: PropTypes.object,
  fetchItems: PropTypes.func.isRequired,
  fetchUpdatedCurrentLoans: PropTypes.func.isRequired,
  clearResults: PropTypes.func.isRequired,
  checkoutItem: PropTypes.func.isRequired,
  patron: PropTypes.number.isRequired,
};

ItemsSearch.defaultProps = {
  queryString: '',
};
