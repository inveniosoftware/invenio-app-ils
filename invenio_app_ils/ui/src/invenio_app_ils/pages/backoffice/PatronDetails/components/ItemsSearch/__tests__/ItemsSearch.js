import React from 'react';
import { shallow, mount } from 'enzyme';
import ItemsSearch from '../ItemsSearch';
import { invenioConfig } from '../../../../../../common/config';
import { invenioConfig as configMock } from '../../../../../../common/__mocks__/config';
import { fromISO } from '../../../../../../common/api/date';
import { Settings } from 'luxon';

jest.mock('../../../../../../common/config');

Settings.defaultZoneName = 'utc';

const stringDate = fromISO('2018-01-01T11:05:00+01:00');

describe('PatronLoans tests', () => {
  let component;
  afterEach(() => {
    if (component) {
      component.unmount();
    }
  });

  invenioConfig['circulation'] = {
    loanActiveStates: configMock.circulation.loanActiveStates,
  };
  invenioConfig['items'] = {
    available: {
      status: configMock.items.available.status,
    },
  };

  it('should load the ItemsSearch component', () => {
    const mockedFetchItems = jest.fn();
    const mockedFetchCurrent = jest.fn();
    const mockedUpdateString = jest.fn();
    const mockedClear = jest.fn();
    const mockedCheckoutItem = jest.fn();

    const component = shallow(
      <ItemsSearch
        items={{ hits: [] }}
        queryString={''}
        updateQueryString={mockedUpdateString}
        fetchItems={mockedFetchItems}
        fetchUpdatedCurrentLoans={mockedFetchCurrent}
        clearResults={mockedClear}
        checkoutItem={mockedCheckoutItem}
        patron={2}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render items in search', () => {
    const mockedFetchItems = jest.fn();
    const mockedFetchCurrent = jest.fn();
    const mockedUpdateString = jest.fn();
    const mockedClear = jest.fn();
    const mockedCheckoutItem = jest.fn();

    const items = {
      hits: [
        {
          id: '1',
          updated: stringDate,
          created: stringDate,
          item_pid: 'item1',
          metadata: {
            document_pid: 'doc1',
            item_pid: 'item1',
            internal_location: { location: { name: 'Somewhere' } },
            barcode: '44444',
            shelf: 'P',
            status: 'CAN_CIRCULATE',
            circulation_status: {},
            medium: 'book',
          },
        },
      ],
      total: 1,
    };

    component = mount(
      <ItemsSearch
        items={items}
        queryString={''}
        updateQueryString={mockedUpdateString}
        fetchItems={mockedFetchItems}
        fetchUpdatedCurrentLoans={mockedFetchCurrent}
        clearResults={mockedClear}
        checkoutItem={mockedCheckoutItem}
        patron={2}
      />
    );

    expect(component).toMatchSnapshot();
    const tableRow = component
      .find('TableRow')
      .filterWhere(
        element => element.prop('data-test') === items.hits[0].item_pid
      );
    expect(tableRow).toHaveLength(1);
  });
});
