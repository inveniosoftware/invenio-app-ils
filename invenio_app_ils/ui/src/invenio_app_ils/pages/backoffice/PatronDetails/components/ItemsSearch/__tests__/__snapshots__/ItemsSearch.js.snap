// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PatronLoans tests should load the ItemsSearch component 1`] = `
<Segment
  className="patron-items"
>
  <Header
    as="h3"
  >
    Items
  </Header>
  <HeaderSubheader>
    Search items by barcode.
  </HeaderSubheader>
  <Container
    className="search-bar"
  >
    <Input
      action={
        Object {
          "content": "Search",
          "onClick": [Function],
        }
      }
      fluid={true}
      focus={true}
      onChange={[Function]}
      onKeyPress={[Function]}
      onPaste={[Function]}
      placeholder="Search by barcode..."
      type="text"
      value=""
    />
  </Container>
  <Grid
    className="items-search-container"
    columns={1}
    relaxed={true}
    stackable={true}
  >
    <GridColumn
      width={16}
    >
      <Loader>
        <Error>
          <Segment
            placeholder={true}
            textAlign="center"
          >
            <Header
              icon={true}
            >
              <Icon
                as="i"
                name="search"
              />
              Found no items matching this barcode.
            </Header>
            <div
              className="empty-results-current"
            >
              Current search phrase "
              "
            </div>
            <SegmentInline>
              <Button
                as="button"
                onClick={[Function]}
                primary={true}
              >
                Clear query
              </Button>
            </SegmentInline>
          </Segment>
        </Error>
      </Loader>
    </GridColumn>
  </Grid>
</Segment>
`;

exports[`PatronLoans tests should render items in search 1`] = `
<ItemsSearch
  checkoutItem={[MockFunction]}
  clearResults={[MockFunction]}
  fetchItems={[MockFunction]}
  fetchUpdatedCurrentLoans={[MockFunction]}
  items={
    Object {
      "hits": Array [
        Object {
          "created": "2018-01-01T10:05:00.000Z",
          "id": "1",
          "metadata": Object {
            "barcode": "44444",
            "circulation": Object {},
            "document": Object {
              "title": "Here is a title",
            },
            "document_pid": "doc1",
            "internal_location": Object {
              "location": Object {
                "name": "Somewhere",
              },
            },
            "medium": "book",
            "pid": "item1",
            "shelf": "P",
            "status": "CAN_CIRCULATE",
          },
          "pid": "item1",
          "updated": "2018-01-01T10:05:00.000Z",
        },
      ],
      "total": 1,
    }
  }
  patronPid="2"
  queryString=""
  updateQueryString={[MockFunction]}
>
  <Segment
    className="patron-items"
  >
    <div
      className="ui segment patron-items"
    >
      <Header
        as="h3"
      >
        <h3
          className="ui header"
        >
          Items
        </h3>
      </Header>
      <HeaderSubheader>
        <div
          className="sub header"
        >
          Search items by barcode.
        </div>
      </HeaderSubheader>
      <Container
        className="search-bar"
      >
        <div
          className="ui container search-bar"
        >
          <Input
            action={
              Object {
                "content": "Search",
                "onClick": [Function],
              }
            }
            fluid={true}
            focus={true}
            onChange={[Function]}
            onKeyPress={[Function]}
            onPaste={[Function]}
            placeholder="Search by barcode..."
            type="text"
            value=""
          >
            <div
              className="ui fluid focus action input"
              onPaste={[Function]}
            >
              <input
                onChange={[Function]}
                onKeyPress={[Function]}
                placeholder="Search by barcode..."
                type="text"
                value=""
              />
              <Button
                as="button"
                content="Search"
                onClick={[Function]}
              >
                <Ref
                  innerRef={
                    Object {
                      "current": <button
                        class="ui button"
                      >
                        Search
                      </button>,
                    }
                  }
                >
                  <RefFindNode
                    innerRef={
                      Object {
                        "current": <button
                          class="ui button"
                        >
                          Search
                        </button>,
                      }
                    }
                  >
                    <button
                      className="ui button"
                      onClick={[Function]}
                    >
                      Search
                    </button>
                  </RefFindNode>
                </Ref>
              </Button>
            </div>
          </Input>
        </div>
      </Container>
      <Grid
        className="items-search-container"
        columns={1}
        relaxed={true}
        stackable={true}
      >
        <div
          className="ui stackable relaxed one column grid items-search-container"
        >
          <GridColumn
            width={16}
          >
            <div
              className="sixteen wide column"
            >
              <Loader>
                <Error>
                  <div
                    className="results-list"
                  >
                    <p>
                      Found 
                      1
                       item(s).
                    </p>
                    <ItemsResultsList
                      checkoutItem={[MockFunction]}
                      clearResults={[MockFunction]}
                      fetchPatronCurrentLoans={[MockFunction]}
                      patronPid="2"
                      results={
                        Object {
                          "hits": Array [
                            Object {
                              "created": "2018-01-01T10:05:00.000Z",
                              "id": "1",
                              "metadata": Object {
                                "barcode": "44444",
                                "circulation": Object {},
                                "document": Object {
                                  "title": "Here is a title",
                                },
                                "document_pid": "doc1",
                                "internal_location": Object {
                                  "location": Object {
                                    "name": "Somewhere",
                                  },
                                },
                                "medium": "book",
                                "pid": "item1",
                                "shelf": "P",
                                "status": "CAN_CIRCULATE",
                              },
                              "pid": "item1",
                              "updated": "2018-01-01T10:05:00.000Z",
                            },
                          ],
                          "total": 1,
                        }
                      }
                    >
                      <ResultsTable
                        columns={
                          Array [
                            Object {
                              "field": "",
                              "formatter": undefined,
                              "title": "",
                            },
                            Object {
                              "field": "metadata.pid",
                              "title": "ID",
                            },
                            Object {
                              "field": "metadata.status",
                              "title": "Status",
                            },
                            Object {
                              "field": "metadata.medium",
                              "title": "Medium",
                            },
                            Object {
                              "field": "metadata.circulation.state",
                              "title": "Circulation status",
                            },
                            Object {
                              "field": "",
                              "formatter": [Function],
                              "title": "Actions",
                            },
                          ]
                        }
                        currentPage={1}
                        data={
                          Array [
                            Object {
                              "created": "2018-01-01T10:05:00.000Z",
                              "id": "1",
                              "metadata": Object {
                                "barcode": "44444",
                                "circulation": Object {},
                                "document": Object {
                                  "title": "Here is a title",
                                },
                                "document_pid": "doc1",
                                "internal_location": Object {
                                  "location": Object {
                                    "name": "Somewhere",
                                  },
                                },
                                "medium": "book",
                                "pid": "item1",
                                "shelf": "P",
                                "status": "CAN_CIRCULATE",
                              },
                              "pid": "item1",
                              "updated": "2018-01-01T10:05:00.000Z",
                            },
                          ]
                        }
                        headerActionComponent={null}
                        name="items"
                        paginationComponent={null}
                        seeAllComponent={null}
                        showAllResults={false}
                        showMaxRows={10}
                        singleLine={false}
                        subtitle=""
                        title=""
                      >
                        <Grid>
                          <div
                            className="ui grid"
                          >
                            <GridRow>
                              <div
                                className="row"
                              >
                                <GridColumn
                                  verticalAlign="middle"
                                  width={16}
                                >
                                  <div
                                    className="middle aligned sixteen wide column"
                                  />
                                </GridColumn>
                                <GridColumn
                                  textAlign="right"
                                  width={8}
                                >
                                  <div
                                    className="right aligned eight wide column"
                                  />
                                </GridColumn>
                              </div>
                            </GridRow>
                          </div>
                        </Grid>
                        <Table
                          as="table"
                          compact={true}
                          selectable={true}
                          striped={true}
                        >
                          <table
                            className="ui selectable striped compact table"
                          >
                            <ResultsTableHeader
                              columns={
                                Array [
                                  Object {
                                    "field": "",
                                    "formatter": undefined,
                                    "title": "",
                                  },
                                  Object {
                                    "field": "metadata.pid",
                                    "title": "ID",
                                  },
                                  Object {
                                    "field": "metadata.status",
                                    "title": "Status",
                                  },
                                  Object {
                                    "field": "metadata.medium",
                                    "title": "Medium",
                                  },
                                  Object {
                                    "field": "metadata.circulation.state",
                                    "title": "Circulation status",
                                  },
                                  Object {
                                    "field": "",
                                    "formatter": [Function],
                                    "title": "Actions",
                                  },
                                ]
                              }
                            >
                              <TableHeader
                                as="thead"
                              >
                                <thead
                                  className=""
                                >
                                  <TableRow
                                    as="tr"
                                    cellAs="td"
                                    data-test="header"
                                  >
                                    <tr
                                      className=""
                                      data-test="header"
                                    >
                                      <TableHeaderCell
                                        as="th"
                                        key=""
                                      >
                                        <TableCell
                                          as="th"
                                          className=""
                                        >
                                          <th
                                            className=""
                                          />
                                        </TableCell>
                                      </TableHeaderCell>
                                      <TableHeaderCell
                                        as="th"
                                        key="ID"
                                      >
                                        <TableCell
                                          as="th"
                                          className=""
                                        >
                                          <th
                                            className=""
                                          >
                                            ID
                                          </th>
                                        </TableCell>
                                      </TableHeaderCell>
                                      <TableHeaderCell
                                        as="th"
                                        key="Status"
                                      >
                                        <TableCell
                                          as="th"
                                          className=""
                                        >
                                          <th
                                            className=""
                                          >
                                            Status
                                          </th>
                                        </TableCell>
                                      </TableHeaderCell>
                                      <TableHeaderCell
                                        as="th"
                                        key="Medium"
                                      >
                                        <TableCell
                                          as="th"
                                          className=""
                                        >
                                          <th
                                            className=""
                                          >
                                            Medium
                                          </th>
                                        </TableCell>
                                      </TableHeaderCell>
                                      <TableHeaderCell
                                        as="th"
                                        key="Circulation status"
                                      >
                                        <TableCell
                                          as="th"
                                          className=""
                                        >
                                          <th
                                            className=""
                                          >
                                            Circulation status
                                          </th>
                                        </TableCell>
                                      </TableHeaderCell>
                                      <TableHeaderCell
                                        as="th"
                                        key="Actions"
                                      >
                                        <TableCell
                                          as="th"
                                          className=""
                                        >
                                          <th
                                            className=""
                                          >
                                            Actions
                                          </th>
                                        </TableCell>
                                      </TableHeaderCell>
                                    </tr>
                                  </TableRow>
                                </thead>
                              </TableHeader>
                            </ResultsTableHeader>
                            <ResultsTableBody
                              columns={
                                Array [
                                  Object {
                                    "field": "",
                                    "formatter": undefined,
                                    "title": "",
                                  },
                                  Object {
                                    "field": "metadata.pid",
                                    "title": "ID",
                                  },
                                  Object {
                                    "field": "metadata.status",
                                    "title": "Status",
                                  },
                                  Object {
                                    "field": "metadata.medium",
                                    "title": "Medium",
                                  },
                                  Object {
                                    "field": "metadata.circulation.state",
                                    "title": "Circulation status",
                                  },
                                  Object {
                                    "field": "",
                                    "formatter": [Function],
                                    "title": "Actions",
                                  },
                                ]
                              }
                              rows={
                                Array [
                                  Object {
                                    "created": "2018-01-01T10:05:00.000Z",
                                    "id": "1",
                                    "metadata": Object {
                                      "barcode": "44444",
                                      "circulation": Object {},
                                      "document": Object {
                                        "title": "Here is a title",
                                      },
                                      "document_pid": "doc1",
                                      "internal_location": Object {
                                        "location": Object {
                                          "name": "Somewhere",
                                        },
                                      },
                                      "medium": "book",
                                      "pid": "item1",
                                      "shelf": "P",
                                      "status": "CAN_CIRCULATE",
                                    },
                                    "pid": "item1",
                                    "updated": "2018-01-01T10:05:00.000Z",
                                  },
                                ]
                              }
                            >
                              <TableBody
                                as="tbody"
                              >
                                <tbody
                                  className=""
                                >
                                  <TableRow
                                    as="tr"
                                    cellAs="td"
                                    data-test="item1"
                                    key="item1"
                                  >
                                    <tr
                                      className=""
                                      data-test="item1"
                                    >
                                      <TableCell
                                        as="td"
                                        data-test="0-item1"
                                        key="0-item1"
                                      >
                                        <td
                                          className=""
                                          data-test="0-item1"
                                        >
                                          -
                                        </td>
                                      </TableCell>
                                      <TableCell
                                        as="td"
                                        data-test="1-item1"
                                        key="1-item1"
                                      >
                                        <td
                                          className=""
                                          data-test="1-item1"
                                        >
                                          item1
                                        </td>
                                      </TableCell>
                                      <TableCell
                                        as="td"
                                        data-test="2-item1"
                                        key="2-item1"
                                      >
                                        <td
                                          className=""
                                          data-test="2-item1"
                                        >
                                          CAN_CIRCULATE
                                        </td>
                                      </TableCell>
                                      <TableCell
                                        as="td"
                                        data-test="3-item1"
                                        key="3-item1"
                                      >
                                        <td
                                          className=""
                                          data-test="3-item1"
                                        >
                                          book
                                        </td>
                                      </TableCell>
                                      <TableCell
                                        as="td"
                                        data-test="4-item1"
                                        key="4-item1"
                                      >
                                        <td
                                          className=""
                                          data-test="4-item1"
                                        >
                                          -
                                        </td>
                                      </TableCell>
                                      <TableCell
                                        as="td"
                                        data-test="5-item1"
                                        key="5-item1"
                                      >
                                        <td
                                          className=""
                                          data-test="5-item1"
                                        >
                                          <Button
                                            as="button"
                                            content="Checkout"
                                            onClick={[Function]}
                                          >
                                            <Ref
                                              innerRef={
                                                Object {
                                                  "current": <button
                                                    class="ui button"
                                                  >
                                                    Checkout
                                                  </button>,
                                                }
                                              }
                                            >
                                              <RefFindNode
                                                innerRef={
                                                  Object {
                                                    "current": <button
                                                      class="ui button"
                                                    >
                                                      Checkout
                                                    </button>,
                                                  }
                                                }
                                              >
                                                <button
                                                  className="ui button"
                                                  onClick={[Function]}
                                                >
                                                  Checkout
                                                </button>
                                              </RefFindNode>
                                            </Ref>
                                          </Button>
                                        </td>
                                      </TableCell>
                                    </tr>
                                  </TableRow>
                                </tbody>
                              </TableBody>
                            </ResultsTableBody>
                            <ResultsTableFooter
                              allRowsNumber={1}
                              columnsNumber={6}
                              currentPage={1}
                              paginationComponent={null}
                              seeAllComponent={null}
                              showAllResults={false}
                              showMaxRows={10}
                            />
                          </table>
                        </Table>
                      </ResultsTable>
                    </ItemsResultsList>
                  </div>
                </Error>
              </Loader>
            </div>
          </GridColumn>
        </div>
      </Grid>
    </div>
  </Segment>
</ItemsSearch>
`;
